spec.html

```diff
@@ -38970,16 +39070,67 @@
         ClassControlLetter ::
           DecimalDigit
           `_`
-
-        ClassAtomNoDash[U] ::
-          SourceCharacter but not one of `\` or `]` or `-`
-          `\` ClassEscape[?U]
-          `\` [lookahead == `c`]
       </emu-grammar>
       <emu-note>
         <p>When the same left hand sides occurs with both [+U] and [\~U] guards it is to control the disambiguation priority.</p>
       </emu-note>
 
+      <emu-annex id="sec-patterns-static-semantics-early-errors-annexb">
+        <h1>Static Semantics: Early Errors</h1>
+        <p>The semantics of <emu-xref href="#sec-patterns-static-semantics-early-errors"></emu-xref> is extended as follows:</p>
+        <emu-grammar>ExtendedAtom :: InvalidBracedQuantifier</emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if any source text matches this rule.
+          </li>
+        </ul>
+        <emu-grammar>NonemptyClassRanges :: ClassAtom `-` ClassAtom ClassRanges</emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if IsCharacterClass of the first |ClassAtom| is *true* or IsCharacterClass of the second |ClassAtom| is *true* <ins>and this production has a [U] parameter</ins>.
+          </li>
```

Comment:
This is what seems to be implemented in V8 and SpiderMonkey; the versions of JSC and ChakraCore I'm testing don't seem to throw the errors in non-Unicode mode.

Reply:
Probably don't want to commit these \<ins\> tags.

Reply:
ins is used elsewhere in Annex B, e.g., [here](https://tc39.github.io/ecma262/#sec-switch-duplicates-allowed-static-semantics).

```
$ eshost -e "/[\w-\b]/u"
#### jsc
SyntaxError: Invalid regular expression: invalid escaped character for unicode pattern

#### chakracore
/[\w-\b]/u

#### spidermonkey
SyntaxError: character class escape cannot be used in class range in regular expression:

#### d8
SyntaxError: Invalid regular expression: /[\w-\b]/: Invalid character class
```